redbeat/schedulers.py

```diff
@@ -191,11 +191,13 @@ def __init__(self, app, **kwargs):
     def setup_schedule(self):
         # cleanup old static entries
         client = redis(self.app)
-        previous = client.smembers(self.app.conf.REDBEAT_STATICS_KEY)
-        current = set(self.app.conf.CELERYBEAT_SCHEDULE.keys())
-        removed = previous - current
+        previous = set(key.decode('utf-8')
+                       for key in client.smembers(self.app.conf.REDBEAT_STATICS_KEY))
+        removed = previous.difference(self.app.conf.CELERYBEAT_SCHEDULE.keys())
         for name in removed:
-            RedBeatSchedulerEntry(name).delete()
+            logger.debug("Removing old schedule entry '%s'.", name)
+            RedBeatSchedulerEntry(name, app=self.app).delete()
+            client.srem(self.app.conf.REDBEAT_STATICS_KEY, name)
```

Comment:
This could leave the schedule information in Redis in a inconsistent state if an exception occurs while deleting a removed static schedule entry. In fact, deleting the schedule entry and the set member should be wrapped in a transaction using `pipeline()` (they can be nested without problems).

I'll update the PR to address this.